autosklearn/automl.py

```diff
@@ -75,38 +75,82 @@
 from autosklearn.util.single_thread_client import SingleThreadedClient
 
 
-def _model_predict(model, X, batch_size, logger, task):
-    def send_warnings_to_log(
-            message, category, filename, lineno, file=None, line=None):
-        logger.debug('%s:%s: %s:%s' % (filename, lineno, category.__name__, message))
-        return
-    X_ = X.copy()
+def _model_predict(
+    model: Any,  # TODO Narrow this down
+    X: SUPPORTED_FEAT_TYPES,
+    task: int,
+    batch_size: Optional[int] = None
+) -> np.ndarray:
+    """ Generates the predictions from a model.
+
+    This is seperated out into a seperate function to allow for multiprocessing
+    and perform parallel predictions.
+
+    # TODO issue 1169
+    #   As CV models are collected into a VotingRegressor which does not
+    #   not support multioutput regression, we need to manually transform and
+    #   average their predictions.
+
+    Parameters
+    ----------
+    model: Any
+        The model to perform predictions with
+
+    X: array-like (n_samples, ...)
+        The data to perform predictions on.
+
+    task: int
+        The int identifier indicating the kind of task that the model was
+        trained on.
+
+    batchsize: Optional[int] = None
+        If the model supports batch_size predictions then it's possible to pass
+        this in as a parameter.
+
+    Returns
+    -------
+    np.ndarray (n_samples, ...)
+        The predictions produced by the model
+    """
+    # Copy the array and ensure is has the attr 'shape'
+    X_ = np.asarray(X) if isinstance(X, list) else X.copy()
+
+    assert X_.shape[0] >= 1, f"X must have more than 1 sample but has {X_.shape[0]}"
+
     with warnings.catch_warnings():
```

Comment:
Is there anything done with the warnings?

Reply:
This line was present from before, I assumed there was some reason it was there.
https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings

I think it is better to remove it as it might suppress information we care to know about

Reply:
IIRC this was there so that the warnings could be swallowed if the user passed `n_jobs>1` to predict(). So I think it would be good to pass warnings to the logger again, as warnings will by default be displayed on the command line.

Reply:
This issue was actually caught in the prior version but instead of throwing an error, it was logged as a warning. This is indeed however a separate case as it would only catch warnings and forward them to the logger them as warnings. It does make me fear the sweep-under-the-rug ability of `logger.warning` though.

If there are none or a small amount of occasional warnings, I would be more inclined to let them be displayed on the command line. I would imagine many users don't check the logs for possible warnings that occur, rather they first see them once there's a problem and we ask them to paste the log. Perhaps letting the user know something is being warned against is useful in case that warning leads to them getting output they were not expecting. 

However, if there are many of the same warnings displayed that no actionable work can be done on then yes I agree it's better to just log them.

Are you familiar with what warnings came up? If not I will reintroduce the logger parameter, log them as a warning and then take a note to occasionally browse them myself to see if anything comes up.

Reply:
So I made the logger optional and it will log to the logger if passed in. For now it's always passed in.